pkg/controllers/multicluster/serviceexport_controller.go

```diff
@@ -164,8 +191,8 @@ func (r *ServiceExportReconciler) handleUpdate(ctx context.Context, serviceExpor
 	return ctrl.Result{}, nil
 }
 
-func (r *ServiceExportReconciler) createOrGetCloudMapService(ctx context.Context, service *v1.Service) (*model.Service, error) {
-	cmService, err := r.CloudMap.GetService(ctx, service.Namespace, service.Name)
+func (r *ServiceExportReconciler) createOrGetCloudMapService(ctx context.Context, service *v1.Service, clustersetId string) (*model.Service, error) {
+	cmService, err := r.CloudMap.GetService(ctx, service.Namespace, service.Name, clustersetId)
```

Comment:
As `clustersetId` is global unique variable, it makes sense to set this globally somewhere instead of passing it around.

Reply:
Added as global variables with additional `refresh` variable. Could of used caching but seems excessive for the use case.